### PR TITLE
Broker and source docs improvements

### LINKED ISSUE #553: Standardize Brokers and Source docs and samples format
**Problem**
When a developer tries the Brokers samples and/or docs, and then wants to try the Source ones, it seems like it is a completely different project, not even the samples have the same format nor the docs.

**[Persona:](https://github.com/knative/eventing/blob/main/docs/personas.md)**
Devs & Contributors

**Exit Criteria**
Both Broker's and Source's docs and samples have the same general structure

### PR DESCRIPTION
**Pull request by gabo1208**
# Changes

- 🎁 New samples docs and files
- 🧹 Updated old source and broker docs (still some work to do but it's a start)

/kind documentation

Fixes #553 

**Release Note**

```release-note
Improved Broker's and Source's Readme docs and Samples description and files.
```

**Docs**

### CODE REVIEW

**Review by xtreme-sameer-vohra — CHANGES_REQUESTED** [review]

File: samples/broker-trigger/README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
-# RabbitMQ Knative Eventing Broker DLQ Example
+# RabbitMQ Knative Eventing Broker with Trigger and DLQ Example
 
 ## Prerequisites
 
-install the Broker as per [here](../../../broker/README.md)
+install the Broker running:
+`kubectl apply -f https://github.com/knative-sandbox/eventing-rabbitmq/releases/latest/download/rabbitmq-broker.yaml`
+
+or as per [here](../../broker/README.md)
```
Comment (xtreme-sameer-vohra) [inline samples/broker-trigger/README.md:9]: Doing `install the Broker running:` won't have the RMQ pre-requisites available, which are required. Does it make sense to link to https://github.com/knative-sandbox/eventing-rabbitmq/blob/main/broker/operator-based.md#rabbitmq-operator-based-knative-eventing-broker ?

File: samples/broker-trigger/README.md [hunk 2]
```diff
@@ -125,20 +132,24 @@ Now the Broker will become ready, might take a few seconds to get the Service up
 and running.
 
 ```sh
-kubectl -n dlq-demo get brokers
+kubectl -n broker-trigger-demo get brokers
 NAME      URL                                                        AGE     READY   REASON
-default   http://default-broker-ingress.dlq-demo.svc.cluster.local   2m39s   True
+default   http://default-broker-ingress.broker-trigger-demo.svc.cluster.local   2m39s   True
```
[the remark this answers is not in the document]
Comment (xtreme-sameer-vohra) [reply samples/broker-trigger/README.md:140]: Thanks for fixing the formatting.
Thoughts on moving `kubectl -n broker-trigger-demo get brokers` to after the Broker install ?

**Review by xtreme-sameer-vohra — CHANGES_REQUESTED** [review]
Thanks @gabo1208 it's looking great :)
Added some minor comments and there are a few comments from before.

File: samples/broker-trigger/README.md [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 ## Prerequisites
 
-install the Broker running:
-`kubectl apply -f https://github.com/knative-sandbox/eventing-rabbitmq/releases/latest/download/rabbitmq-broker.yaml`
+Same as listed [here](../../broker/README.md#prerequisites)
```
Comment (xtreme-sameer-vohra) [inline samples/broker-trigger/README.md:5]:
```suggestion
Same as listed [here](../../broker/README.md)
```

`#prerequisites` doesn't appear to be defined 🤔

File: samples/broker-trigger/README.md [hunk 4]
```diff
@@ -2,10 +2,10 @@
...
 
-or as per [here](../../broker/README.md)
+Now install the Broker running:
```
Comment (xtreme-sameer-vohra) [inline samples/broker-trigger/README.md:7]: The previous step points to the [installation](https://github.com/gabo1208/eventing-rabbitmq/blob/broker-and-source-docs-improvements/broker/operator-based.md#installation) which would include installing the rabbitmq-broker.yaml
Is this step required ?

File: samples/source/README.md [hunk 5]
```diff
@@ -2,11 +2,11 @@
 
 ## Prerequisites
 
+Same as listed [here](../../source/DEVELOPMENT.md#prerequisites)
```
Comment (xtreme-sameer-vohra) [inline samples/source/README.md:5]: What are the pre-requisites from the development side required for trying this sample ?

File: samples/source/README.md [hunk 6]
```diff
@@ -2,11 +2,11 @@
...
+
 install the Source running:
```
Comment (xtreme-sameer-vohra) [inline samples/source/README.md:7]:
```suggestion
Install the Source by running:
```

**Review by xtreme-sameer-vohra — CHANGES_REQUESTED** [review]
It would be valuable for the samples to simplify the setup required to run them, so does it make sense to refer to the releases of the Broker and Source and remove references to building them from source code ?

File: broker/README.md [hunk 7]
```diff
@@ -1,23 +1,32 @@
...
+RabbitMQ *is a Messaging Broker* - an intermediary for messaging. It gives your applications a common platform to send and receive messages, and your messages a safe place to live until received.
+
 ![RabbitMQ Broker for Knative Eventing](rabbitmq-knative-broker.png)
 
+# Table of Contents
 
+- [Installation](#installation)
+  - [Standalone Broker](#standalone-broker)
+  - [RabbitMQ Operator Based Broker](#rabbitmq-operator-based-broker)
+- [Autoscaling](#autoscaling)
```
Comment (xtreme-sameer-vohra) [inline broker/README.md:12]:
```suggestion
```
Dead link

File: source/README.md [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+
+The Source will provide output information about readiness or errors via the
+`status` field on the object once it has been created in the cluster.
+
+<!---// TODO: should we have error documentation?--->
+
+### Configuration Options
+
+- Event source parameters.
+  - Configure channel config properties based on this documentation.
```
Comment (gabo1208) [inline source/README.md:235]:
```suggestion

  - Configure channel config properties based on this documentation.
```

**Comment by xtreme-sameer-vohra** [comment]
Thanks for driving this @gabo1208  👍

**Review by xtreme-sameer-vohra — APPROVED** [review]

**Pull request merged**